src/graphnet/data/dataloader.py

```diff
@@ -63,23 +63,51 @@ def from_dataset_config(
                 "`shuffle` is automatically inferred from the selection name, "
                 "and thus should not specified as an argument."
             )
-            datasets = Dataset.from_config(config)
+
+            if isinstance(config.path, list):
+                datasets: Union[Dict[str, Dataset], Dict[str, EnsembleDataset]] = {}  # type: ignore
```

Comment:
I think the reason for your problem with mypy could be that you define `datasets` to contain both `Dataset` and `EnsembleDataset`, but in the typehints of `DataLoader` the `dataset` is only type hinted to be `dataset: Dataset`. See [here](https://github.com/graphnet-team/graphnet/blob/d0d987474f1f4e76e53384b46a9d1950d3672d56/src/graphnet/data/dataloader.py#L31). I think if you changed that type hint to `dataset: Union[Dataset, EnsembleDataset]`, the type hinting should be fine. Did you try this?